### DISCUSSION TITLE
connected words

### DISCUSSION THREAD

**Discussion by 88arvin** [explosion/spaCy#13000] [Help: Other Questions] [answered]
I have trained a custom NER spacy model for extracting names from the ABC column to the NAME column, but I am having one issue, i.e., some names in the ABC column are connected with other words, so my spacy NER model is not extracting that name. Any ideas on how to extract those words? Although training data includes examples of names that are connected to other words.

**Comment by danieldk** [comment]
Could you give examples of what the training inference data looks like?

**Reply by 88arvin** [reply]
742203200000000000RESEARCHBYTES CORPORAT 99922
00000EQUITAS SMALL FINANCAT
1823551MOTHERS SCHOOL
2768:ASHOKCORP00004

**Reply by danieldk** [reply]
It is still not clear to me from the example what the docs are and what the label (and what the ABC column and what the NAME column)? I also wonder if spaCy is the right tool for this task, this doesn't really look like natural language text, but I guess it's worth a try.

**Reply by 88arvin** [reply]
I have a Pandas dataframe with a column called ABC that has the following types of values: And I'd like to extract the names in a new column called NAME.
 
`742203200000000000RESEARCHBYTES CORPORAT 99922` ------`RESEARCHBYTES CORPORAT`
`00000EQUITAS SMALL FINANCAT` ----------------------------`EQUITAS SMALL FINANCAT`
`1823551MOTHERS SCHOOL` -----------------------------------`MOTHERS SCHOOL`
`2768:ASHOKCORP00004` -------------------------------------`ASHOKCORP`

As a result, to extract the names, I trained a custom NER spacy model. Although I can extract the names, some names in the ABC column are connected to other words, which my custom NER spacy model cannot extract.

**Reply by danieldk** [reply, marked as answer]
Thanks for elaborating! I see what you mean now. In these cases it's best to add exceptions to the tokenizer (or even a custom tokenizer) to improve tokenization of your data. See https://spacy.io/usage/linguistic-features#special-cases .
